### PR TITLE
[ENH] Port RDST for regression

### PR DESCRIPTION
**Pull request by MatthewMiddlehurst**
Creates a regressor version of RDST. Last of the simple pipeline classifiers to be ported for now.

### CODE REVIEW

File: aeon/regression/shapelet_based/_rdst.py [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+    def __init__(
+        self,
+        max_shapelets=10000,
+        shapelet_lengths=None,
+        proba_normalization=0.8,
+        threshold_percentiles=None,
+        alpha_similarity=0.5,
+        use_prime_dilations=False,
+        estimator=None,
+        save_transformed_data=False,
```
Comment (baraline) [inline aeon/regression/shapelet_based/_rdst.py:129]: `save_transformed_data` does not appear in the class parameter doc

**Review by baraline — COMMENTED** [review]
Apart from these two minor documentation points, LGTM ! Thanks for putting it together 😅

**Comment by MatthewMiddlehurst** [comment]
Fixed for classification as well

**Review by baraline — APPROVED** [review]
Nice catch for the classification one ! LGTM.

**Pull request merged**